### PR TITLE
nagelfar: init at 1.3.3

### PR DESCRIPTION
**Pull request by nat-418**
###### Description of changes

Nagelfar is a static syntax checker for the Tcl programming language.

Homepage: https://nagelfar.sourceforge.net/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by nat-418** [comment]
@GrahamcOfBorg build nagelfar

File: pkgs/development/tools/nagelfar/default.nix
Comment (K900) [inline pkgs/development/tools/nagelfar/default.nix:17]: This is pretty inconsistent with the code style used in the rest of nixpkgs.

File: maintainers/maintainer-list.nix
Comment (K900) [inline maintainers/maintainer-list.nix:9543]: Should be a separate commit.

**Comment by NobbZ** [comment]
Result of `nixpkgs-review pr 199106` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Comment by NobbZ** [comment]
Result of `nixpkgs-review pr 199106` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)

File: maintainers/maintainer-list.nix
Comment (nat-418) [reply maintainers/maintainer-list.nix:9543]: done

File: pkgs/development/tools/nagelfar/default.nix
Comment (nat-418) [reply pkgs/development/tools/nagelfar/default.nix]: updated

**Comment by nat-418** [comment]
> Result of `nixpkgs-review pr 199106` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)

What does this mean?

**Comment by NobbZ** [comment]
That nothing was built, neither on x86-64 nor on aarch64.

**Review by NobbZ — CHANGES_REQUESTED** [review]
* Please also add the new package to `pkgs/top-level/all-packages.nix` such that it can be built… Traditionally in the same commit that also adds the implementing file.
* Please reorder your commits, such that the maintainer exists before you use it

**Comment by nat-418** [comment]
@NobbZ let me know if this looks better to you. Thanks for the help.

**Comment by NobbZ** [comment]
Would you mind fixing the conflict?

I want to build and test it. I don't know much about `tcl`, though as there is literally no build happening and the `buildInputs` neither used as far as I can tell and no references created, I doubt that this works as intended…

**Comment by nat-418** [comment]
@NobbZ Conflict fixed. I build Nagelfar using `nix-build -E 'with import <nixpkgs> {}; callPackage ./default.nix {}'` and it works fine. Is there something I am missing here?

**Comment by NobbZ** [comment]
Result of `nixpkgs-review pr 199106` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nagelfar</li>
  </ul>
</details>

**Comment by NobbZ** [comment]
Result of `nixpkgs-review pr 199106` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nagelfar</li>
  </ul>
</details>

**Review by NobbZ — APPROVED** [review]
I checked and indeed it works... Checking the output, it seems to be wrapped. I don't understand the how or why, but I assume it's tcl specific.

**Comment by nat-418** [comment]
> I checked and indeed it works... Checking the output, it seems to be wrapped. I don't understand the how or why, but I assume it's tcl specific.

I used other Tcl applications in nixpkgs like tcl2048 as templates for Nagelfar. This is my first time contributing a package to the Nix ecosystem, thanks again for your help.

File: pkgs/development/tools/nagelfar/default.nix
Comment (K900) [inline pkgs/development/tools/nagelfar/default.nix:23]: Can get away with `install -Dm`

File: pkgs/development/tools/nagelfar/default.nix
Comment (K900) [inline pkgs/development/tools/nagelfar/default.nix:18]: Can also replace all of those with `buildCommand` probably.

File: pkgs/top-level/all-packages.nix
Comment (K900) [inline pkgs/top-level/all-packages.nix:38601]: Should probably go somewhere around alphabetical order.

File: pkgs/development/tools/nagelfar/default.nix
Comment (K900) [inline pkgs/development/tools/nagelfar/default.nix:1]: `runtimeShell` seems unused

Comment (K900) [reply pkgs/development/tools/nagelfar/default.nix:1]: And same for `stdenv` actually

File: pkgs/development/tools/nagelfar/default.nix
Comment (nat-418) [reply pkgs/development/tools/nagelfar/default.nix]: That results in a broken build in my testing: Nagelfar is both a graphical and command-line application, and putting installPhase in buildCommand renders the GUI inoperable.

File: pkgs/development/tools/nagelfar/default.nix
Comment (nat-418) [reply pkgs/development/tools/nagelfar/default.nix]: removed

**Comment by nat-418** [comment]
@K900 looks like my PR has passed the checks and review from you and @NobbZ. What else do I need to do to get this PR merged?

**Comment by K900** [comment]
Thanks for the ping - I was going to merge it when ofborg was done, but got sidetracked by real life stuff.

**Comment by NobbZ** [comment]
And I do never merge stuff I have approved by myself. Thats the workflow I am used to from work…

**Comment by nat-418** [comment]
Thank you both for your help with my first Nix package contribution.

**Pull request merged**
